app/src/main/java/com/github/fribourgsdp/radio/game/Game.kt

```diff
@@ -282,5 +328,14 @@ class Game private constructor(val id: Long, val name: String, val host: User, v
             }
         }
 
+        /**
+         * Predicate that is true for songs that have lyrics, and false for songs whose lyrics are not defined due to
+         * - error in the backend
+         * - lyrics nonexistence
+         */
+        val songHasLyrics : (Song) -> Boolean = { song : Song ->
+            !(song.lyrics == MusixmatchLyricsGetter.BACKEND_ERROR_PLACEHOLDER || song.lyrics == MusixmatchLyricsGetter.LYRICS_NOT_FOUND_PLACEHOLDER)
+        }
```

Comment:
A little weird for this to be in a companion object for the game instead of being an actual method of song

Reply:
I agree with @shanzhaiii that this should be a method in song (and also not a lambda).

Reply:
I agree it would be better in Song.kt. Why not a lambda by the way ?

Reply:
I see, it can be in the class Song itself, not the companion object, and simply returning a boolean.